### PR TITLE
Update Package Generator for NIM.

### PR DESCRIPTION
**Pull request by pszczesniak**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature (generator): New flag `--installation-methods` that should allow to generate package with current installation methods of CKEditor 5 or with current and legacy methods with DLLs. See https://github.com/ckeditor/ckeditor5/issues/15502, https://github.com/ckeditor/ckeditor5/issues/15739

---

⚠️  Please do not merge after approved review. ⚠️ 

ℹ️  **After approved review we would like to release alpha version - to make it available for tests.** ℹ️ 

### Additional information

- In templates that are using New Installation Methods `ckeditor5` and `ckeditor5-premium-features` are using `nightly`,

### Supported scopes

* `generator` → https://www.npmjs.com/package/ckeditor5-package-generator
* `*` → https://www.npmjs.com/package/@ckeditor/ckeditor5-package-*

### CODE REVIEW

**Review by filipsobol — APPROVED** [review]

**Review by przemyslaw-zan — CHANGES_REQUESTED** [review]
The `build:dist` script in non-legacy TS package using yarn shows some kind of warning:

```
$ yarn run  build:dist
yarn run v1.22.19
$ node ./scripts/build-dist.mjs
1/2: Generating NPM build...
2/2: Generating browser build...
No name was provided for external module "ckeditor5" in "output.globals" – guessing "ckeditor5".
Done in 5.36s.
```

Other than that, everything seems to work properly.

File: packages/ckeditor5-package-generator/lib/templates/ts/tsconfig.dist.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
```
[the remark this answers is not in the document]
Comment (pszczesniak) [reply packages/ckeditor5-package-generator/lib/templates/ts/tsconfig.dist.json:1]: Will remove it from here and will mention about it in docs for `package-generator` and in migration guide.

File: packages/ckeditor5-package-generator/lib/utils/copy-files.js [hunk 2]
```diff
@@ -68,7 +71,7 @@ function copyTemplate( templatePath, packagePath, data ) {
 
 	const processedTemplatePath = templatePath
 		// Remove sub-directory inside templates to merge results into one directory.
-		.replace( /^(?:common|js|ts)(?:\\|\/)/, '' )
+		.replace( /^(?:common|js|ts|legacy-js|legacy-ts)(?:\\|\/)/, '' )
```
[the remark this answers is not in the document]
Comment (pszczesniak) [reply packages/ckeditor5-package-generator/lib/utils/copy-files.js]: Fully agree. Addressed in 9b05427

**Comment by pszczesniak** [comment]
> The `build:dist` script in non-legacy TS package using yarn shows some kind of warning:

Yes, it's on out todo list to fix in `ckeditor5-dev-build-tools` it will be done till the `CKEditor5` release with NIM.

**Review by przemyslaw-zan — APPROVED** [review]
LGTM 👍

**Review by przemyslaw-zan — CHANGES_REQUESTED** [review]
I think we should rename the cli option, to something like:
`--without-legacy-methods` => `--use-new-installation-methods`

**Comment by pszczesniak** [comment]
@przemyslaw-zan we decided to name the flag `--use-only-new-installation-methods` to make it more explicit.

**Comment by pomek** [comment]
Changes from this branch have been released: https://github.com/ckeditor/ckeditor5-package-generator/releases/tag/v2.0.0-alpha.0.

**Pull request merged**